xnat_admin_tools/create_new_projects.py

```diff
@@ -86,32 +86,48 @@ def set_xsync_credentials(
         response["estimatedExpirationTime"],
     )
 
-    get_url = xserver_host + f"/data/projects/{project_id}/config/xsync"
+    get_url = xrelay_host + f"/data/projects/{project_id}/config/xsync"
     R = requests.request(
         "GET",
         get_url,
         headers={"Content-Type": "application/json"},
         auth=basic,
     )
-    response = R.json()
 
-    xsync_config = response["ResultSet"]["Result"][0]["contents"]
-    content_dict = json.loads(xsync_config)
-    remote_project_id = content_dict["remote_project_id"]
-    # make the post call to xsync on relay
-    basic = HTTPBasicAuth(xrelay_user, xrelay_pass)
-    payload = {
-        "username": xserver_user,
-        "secret": secret,
-        "host": xserver_host,
-        "alias": alias,
-        "localProject": project_id,
-        "remoteProject": remote_project_id,
-        "estimatedExpirationTime": expiration,
-    }
+    # Project has XSync Configuration. Fetch Remote Project ID for Payload
+    if R.status_code == 200:
+        response = R.json()
+        xsync_config = response["ResultSet"]["Result"][0]["contents"]
+        content_dict = json.loads(xsync_config)
+        remote_project_id = content_dict["remote_project_id"]
+
+        # make the post call to xsync on relay
+        basic = HTTPBasicAuth(xrelay_user, xrelay_pass)
+        payload = {
+            "username": xserver_user,
+            "secret": secret,
+            "host": xserver_host,
+            "alias": alias,
+            "localProject": project_id,
+            "remoteProject": remote_project_id,
+            "estimatedExpirationTime": expiration,
+        }
+    # Project has no XSync Configuration. New Project. Use Project ID for Remote Project for Payload
```

Comment:
I assumed from the code that the configuration file is setup in `set_project_settings()`, which is called after `set_xsync_credentials()` in `create_new_projects()`.  If that is the case, a config file won't exist for new projects when `set_xsync_credentials()` is called. 

Maybe a better solution is to rearrange these function calls in `create_new_projects()`, so that the configuration settings are applied before setting xsync credentials which depend on them?

Reply:
Ah, I see. Re-arranging the logic makes sense to me